src/compiler/crystal/tools/doc/macro.cr

```diff
@@ -29,7 +29,7 @@ class Crystal::Doc::Macro
 
   def id
     String.build do |io|
-      io << to_s.gsub(/<.+?>/, "").gsub(' ', "")
+      io << to_s.gsub(/<.+?>/, "").delete(' ')
       io << "-macro"
```

Comment:
`io << to_s.gsub(/<.+?>/, "").delete(' ') << "-macro"` on a single line

Reply:
you can also do
```crystal
io << to_s.gsub(/<.+?>/, "").delete(' ') \
    << "-macro"
```
to continue the appends to `io` and not create a new one.

Reply:
nvm, the performance here doesn't really matters as @straight-shoota said.

Reply:
There's no performance benefit in that I don't see one style wise either.

Reply:
@j8r Please, stop with those style-related comments. This PR is about perf improvements, not specific code style preferences.

Reply:
here are the results
```crystal
require "benchmark"

Benchmark.ips do |x|
  x.report "one append to io" do
    999.times do
    String.build do |io|
      io << "abc" << "bcd"
    end
    end
  end

  x.report "one append to io with backslash" do
    999.times do
    String.build do |io|
      io << "abc" \
         << "bcd"
    end
    end
  end

  x.report "two appends to io" do
    999.times do
    String.build do |io|
      io << "abc"
      io << "bcd"
    end
    end
  end
end
```
results:
```
               one append to io   7.59k (131.82µs) (± 9.25%)  1.10× slower
one append to io with backslash   8.37k (119.44µs) (± 2.14%)       fastest
              two appends to io   8.37k (119.49µs) (± 2.43%)  1.00× slower
```
I'm wrong on which one is fast @Sija, but you're wrong saying that there are no speed differences.

Reply:
There's no difference, `131.82 * 0.092 = 12.12`, `131.82 - 12.12 = 119.7

Reply:
Your benchmark result is clearly wrong because the first two examples produce exactly the same AST and byte code. If one is faster than the other, that's because of external influences.

Reply:
They're all the same even in the results, no point in even arguing about environmental issues these micro benchmarks have.